### PR TITLE
Updating FunctionsNetHost to handle the new "functions-" prefix cmdline args.

### PR DESCRIPTION
**Pull request by kshyju**
Updating FunctionsNetHost to handle the new "functions-" prefix cmdline args.

Part of #1900  

Also bumped up the global JSON .net version to rc1.

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [ ] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: host/src/FunctionsNetHost/Program.cs [hunk 1]
```diff
@@ -35,25 +34,36 @@ private static async Task<GrpcWorkerStartupOptions> GetStartupOptionsFromCmdLine
...
+            rootCommand.SetHandler((context) =>
             {
-                workerStartupOptions.Host = host;
-                workerStartupOptions.Port = port;
-                workerStartupOptions.WorkerId = workerId;
-                workerStartupOptions.GrpcMaxMessageLength = grpcMsgLength;
-                workerStartupOptions.RequestId = requestId;
-            },
-                hostOption, portOption, workerOption, grpcMsgLengthOption, requestIdOption);
+                var serverEndpoint = context.ParseResult.GetValueForOption(funcUriOption) ?? $"http://{context.ParseResult.GetValueForOption(hostOption)}:{context.ParseResult.GetValueForOption(portOption)}";
```
Comment (kshyju) [inline host/src/FunctionsNetHost/Program.cs:60]: Synced with Jon from the .net command line team and he suggested I should switch to the new approach which uses the `InvocationContext` and calling the `GetValueForOption` method. This is something they have introduced in the newer version.

> This is closer to the pattern we're arriving at in the next version. Handler parameter binding is going away as a concept.

File: host/src/FunctionsNetHost/Program.cs [hunk 2]
```diff
@@ -35,25 +34,36 @@ private static async Task<GrpcWorkerStartupOptions> GetStartupOptionsFromCmdLine
...
+                workerStartupOptions.ServerUri = new Uri(serverEndpoint);
```
Comment (fabiocav) [inline host/src/FunctionsNetHost/Program.cs:61]: Can we use `TryCreate` and provide a clear error if we have an invalid URI?

File: host/src/FunctionsNetHost/Program.cs [hunk 3]
```diff
@@ -35,25 +34,36 @@ private static async Task<GrpcWorkerStartupOptions> GetStartupOptionsFromCmdLine
...
+
+                workerStartupOptions.GrpcMaxMessageLength = context.ParseResult.GetValueForOption(funcGrpcMsgLengthOption) ?? context.ParseResult.GetValueForOption(grpcMsgLengthOption);
```
Comment (fabiocav) [inline host/src/FunctionsNetHost/Program.cs:63]: As discussed, shouldn't need the fallback checks.

**Review by fabiocav — APPROVED** [review]

File: host/src/FunctionsNetHost/Program.cs (hunk 2 above)
Comment (kshyju) [reply host/src/FunctionsNetHost/Program.cs]: The new iteration uses TryCreate to validate and throw a more meaningful message.

File: host/src/FunctionsNetHost/Program.cs (hunk 3 above)
Comment (kshyju) [reply host/src/FunctionsNetHost/Program.cs]: Good point. Removed fallback.

**Review by jviau — APPROVED** [review]

**Pull request merged**